### DISCUSSION TITLE
Challenges in plotting IPF and KAM maps

### DISCUSSION THREAD

**Discussion by Sogo5030** [mtex-toolbox/mtex#2149] [Ask Anything]
I am new to MTEX and trying to get my IPF and KAM maps but my IPF map gives a blank figure.

I don't know what I am doing wrong. can someone give me an idea?
```Matlab
%% Import Script for EBSD Data
%
% This script was automatically created by the import wizard. You should
% run the whoole script or parts of it in order to import your data. There
% is no problem in making any changes to this script.

%% Specify Crystal and Specimen Symmetries

% crystal symmetry
CS = {... 
  'notIndexed',...
  crystalSymmetry('m-3m', [2.9 2.9 2.9], 'mineral', 'Ferrite, bcc (New)', 'color', [0.53 0.81 0.98])};

% plotting convention
setMTEXpref('xAxisDirection','north');
setMTEXpref('zAxisDirection','outOfPlane');

%% Specify File Names

% path to files
pname = 'C:\Users\Desktop\Code\EBSD\MAp 43';

% which files to be imported
fname = [pname '\map4.ctf'];

%% Import the Data

% create an EBSD variable containing the data
ebsd = EBSD.load(fname,CS,'interface','ctf',...
  'convertEuler2SpatialReferenceFrame');
if isa(ebsd, 'EBSD')
    ipfkey = ipfTSLKey(ebsd);
    plot(ipfkey);
    %plot(ebsd, ipfkey.orientation2color(ebsd.orientations));
    %colorbar;
    %title('Inverse Pole Figure Map');
else
    error('Invalid EBSD data object');
end
```

**Comment by kilir** [comment]
Hi,

what does `ebsd('f')` return.

If there is some data in this object, can you try ?
```Matlab
ipfkey = ipfHSVKey(ebsd('f'));
plot(ebsd('f'), ipfkey.orientation2color(ebsd('f').orientations))
```

Cheers,

Rüdiger

**Reply by Sogo5030** [reply]
Thank you for your kind response. I tried it but my figure popped up with no distribution.

Here is what I did:

%% Import Script for EBSD Data
%
% This script was automatically created by the import wizard. You should
% run the whoole script or parts of it in order to import your data. There
% is no problem in making any changes to this script.

%% Specify Crystal and Specimen Symmetries

% crystal symmetry
CS = {... 
  'notIndexed',...
  crystalSymmetry('m-3m', [2.9 2.9 2.9], 'mineral', 'Ferrite, bcc (New)', 'color', [0.53 0.81 0.98])};

% plotting convention
setMTEXpref('xAxisDirection','north');
setMTEXpref('zAxisDirection','outOfPlane');

%% Specify File Names

% path to files
pname = 'C:\Users\Desktop\Code\EBSD\MAp 43';

% which files to be imported
fname = [pname '\
![Fig](https://github.com/mtex-toolbox/mtex/assets/171449704/8d0d2507-1569-4757-b243-bff0f66b7822)
map4.ctf'];

%% Import the Data

% create an EBSD variable containing the data
ebsd = EBSD.load(fname,CS,'interface','ctf',...
  'convertEuler2SpatialReferenceFrame');

%% rotate the orientation data but not the spatial data

rot=rotation('Euler', 90*degree, 0*degree, 0*degree);
ebsd=rotate(ebsd,rot,'keepXY'); % 'keepXY' allows you to apply a rotation to your orientation data without changing the spatial data

RD = vector3d.X;
TD = vector3d.Y;
ND = vector3d.Z;

ipfKey = ipfTSLKey(ebsd('Ferrite, bcc (New)'));
figure(1);
ipfKey.inversePoleFigureDirection = xvector;
oricolors = ipfKey.orientation2color(ebsd('Ferrite, bcc (New)').orientations);
plot(ebsd('Ferrite, bcc (New)'),oriColors,'figSize','small')
hold on
% plot(grains.boundary,'linewidth',1)
title('IPF X')
hold off


% Plot IPF map
%plot(ebsd, colors);
%colorbar;
%title('IPF-Z Map');
